lib/racecar/daemon.rb

```diff
@@ -0,0 +1,91 @@
+module Racecar
+  class Daemon
+    attr_reader :pidfile
+
+    def initialize(pidfile)
+      @pidfile = pidfile
+    end
+
+    def pidfile?
+      !pidfile.nil?
+    end
+
+    def daemonize!
+      exit if fork
+      Process.setsid
+      exit if fork
+      Dir.chdir "/"
+    end
+
+    def redirect_output(logfile)
+      FileUtils.mkdir_p(File.dirname(logfile), :mode => 0755)
+      FileUtils.touch logfile
+      File.chmod(0644, logfile)
+      $stderr.reopen(logfile, 'a')
+      $stdout.reopen($stderr)
+      $stdout.sync = $stderr.sync = true
+    end
+
+    def suppress_output
+      $stderr.reopen('/dev/null', 'a')
+      $stdout.reopen($stderr)
+    end
+
+    def check_pid
+      if pidfile?
+        case pid_status
+        when :running, :not_owned
+          $stderr.puts "=> Racecar is already running with that PID file (#{pidfile})"
+          exit(1)
+        when :dead
+          File.delete(pidfile)
+        end
+      end
+    end
+
+    def pid
+      if File.exists?(pidfile)
+        File.read(pidfile).to_i
+      else
+        nil
+      end
+    end
+
+    def running?
+      pid_status == :running || pid_status == :not_owned
+    end
+
+    def stop!
+      Process.kill("TERM", pid)
+    end
+
+    def pid_status
+      return :exited if pid.nil?
+      return :dead if pid == 0
+
+      # check process status
+      Process.kill(0, pid)
```

Comment:
Will update the comment – it just raises an exception if the process doesn't exist.